include/irr/builtin/material_compiler/glsl/common_declarations.glsl

```diff
@@ -0,0 +1,44 @@
+#ifndef _IRR_BUILTIN_MATERIAL_COMPILER_GLSL_COMMON_DECLARATIONS_INCLUDED_
+#define _IRR_BUILTIN_MATERIAL_COMPILER_GLSL_COMMON_DECLARATIONS_INCLUDED_
+
+#include <irr/builtin/glsl/virtual_texturing/extensions.glsl>
+#include <irr/builtin/glsl/colorspace/encodeCIEXYZ.glsl>
+
+#define instr_t uvec2
+#define reg_t uint
+#define params_t mat4x3
+#define bxdf_eval_t vec3
+#define eval_and_pdf_t vec4
+
+#define INSTR_1ST_DWORD(instr) (instr).x
+#define INSTR_2ND_DWORD(instr) (instr).y
+
+struct bsdf_data_t
+{
+	uvec4 data[sizeof_bsdf_data];
```

Comment:
where's the definition of `sizeof_bsdf_data` ?

Reply:
generated dynamically in `CMaterialCompilerGLSLBackendCommon::genPreprocDefinitions()` so that GLSL is always up to date with C++ constants without editing GLSL by hand

Reply:
ah cool.